lms/djangoapps/courseware/module_render.py

```diff
@@ -1040,6 +1040,9 @@ def handle_xblock_callback(request, course_id, usage_id, handler, suffix=None):
                     request.user, _ = user_auth_tuple
                     break
 
+    if not request.user.is_authenticated:
+        return HttpResponse('Unauthenticated', status=403)
+
```

Comment:
This code is now in conflict with the code right below it!  The code below is trying to ensure that GETs are still allowed for anonymous users.  We need to raise a 403 with the "Unauthenticated" string only for non-GET requests in order to be feature compliant.

So, we need to move this change down into the `HttpResponseForbidden()` call 4 lines below.

Reply:
@ormsbee Does this change break the new Anonymous Courseware access feature that we're trying to roll out for Ironwood?  Or is the new implementation orthogonal to the initial one from 2017?

Reply:
@nasthagiri: I think it will. Need to dig more to know for sure. Can we revert for now?

Reply:
If we need to revert this PR for now, let's do so, I will reach out to @uzairr to see if he is still available to do so.